File: R/ex3.R
#!/bin/Rscript

## Exercício 1.8
main <- function()
{
  cat("Entre com o texto para a codificação: ")
  string <- as.character(readLines("stdin", n = 1))

  codificar(string)
}

codificar <- function(string)
{
  lista <- strsplit(string, "")[[1]]

  if(length(lista)%%2 == 1)
  {
    n <- length(lista)+1
    lista <- c(lista, "#")
  } else
  {
    n <- length(lista)
  }

  lista_n <- c()
  
  for(i in 1:n)
  {
    if(i%%2 == 0)
    {
      lista_n <- c(lista_n, i)
    }
  }
  
  text <- ""
  
  for(i in lista_n)
  {
    text <- paste0(text,
                   as.character(lista[i]),
                   as.character(lista[i-1]))
  }
  
  return(text)
}

main()

File: py/ex3.py
#!/bin/python

def main():
  string = str(input("Entre com o texto para codificação: "))
  codificar(string)

  
def codificar(string):
  lista = list(string)
  
  if len(lista)%2 == 1:
    n = len(lista)+1
    lista.append("#")
  else:
    n = len(lista)
  
  lista_n = []
    
  for i in range(1, n+1):
    if i%2 == 1:
      lista_n.append(i)
      
  text = ""
    
  for i in lista_n:
    text = text+str(lista[i])+str(lista[i-1])
  
  print(text)
  
if __name__ == "__main__":
  main()
